tests/src/kakarot/accounts/test_account_contract.py

```diff
@@ -351,176 +348,508 @@ def test_should_execute_starknet_call(self, cairo_run):
                 calldata=calldata,
             )
 
-    class TestValidate:
-        @pytest.mark.parametrize("seed", (41, 42))
-        @pytest.mark.parametrize("transaction", TRANSACTIONS)
+    class TestExecuteFromOutside:
         @SyscallHandler.patch(
-            "Account_cairo1_helpers_class_hash", CAIRO1_HELPERS_CLASS_HASH
+            "IKakarot.get_cairo1_helpers_class_hash",
+            lambda addr, data: [CAIRO1_HELPERS_CLASS_HASH],
         )
-        def test_should_pass_all_transactions_types(self, cairo_run, seed, transaction):
-            """
-            Note: the seeds 41 and 42 have been manually selected after observing that some private keys
-            were making the Counter deploy transaction failing because their signature parameters length (s and v)
-            were not 32 bytes.
-            """
-            random.seed(seed)
+        @pytest.mark.parametrize("transaction", TRANSACTIONS)
+        def test_should_raise_with_incorrect_signature_length(
+            self, cairo_run, transaction
+        ):
             private_key = generate_random_private_key()
             address = int(private_key.public_key.to_checksum_address(), 16)
             signed = Account.sign_transaction(transaction, private_key)
+            signature = [
+                *int_to_uint256(signed.r),
+                signed.s,
+                signed["v"],
+            ]
+            encoded_unsigned_tx = rlp_encode_signed_data(transaction)
+            tx_data = list(encoded_unsigned_tx)
 
+            with cairo_error(
+                message="Incorrect signature length"
+            ), SyscallHandler.patch("Account_evm_address", address):
+                cairo_run(
+                    "test__execute_from_outside",
+                    tx_data=tx_data,
+                    signature=signature,
+                    chain_id=transaction.get("chainId") or CHAIN_ID,
+                )
```

Comment:
make the example minimal


```suggestion
            with cairo_error(
                message="Incorrect signature length"
            ), SyscallHandler.patch("Account_evm_address", address):
                cairo_run(
                    "test__execute_from_outside",
                    tx_data=[],
                    signature=list(range(4)),
                    chain_id=CHAIN_ID,
                )
```